macOS/DuckDuckGo/MainWindow/MainViewController.swift

```diff
@@ -628,35 +630,48 @@ extension MainViewController {
                 isHandled = isHandled || addressBarVC.escapeKeyDown()
             }
             return isHandled
+        }
 
-        // Handle critical Main Menu actions before WebView
-        case kVK_ANSI_1, kVK_ANSI_2, kVK_ANSI_3, kVK_ANSI_4, kVK_ANSI_5, kVK_ANSI_6,
-             kVK_ANSI_7, kVK_ANSI_8, kVK_ANSI_9,
-             kVK_ANSI_Keypad1, kVK_ANSI_Keypad2, kVK_ANSI_Keypad3, kVK_ANSI_Keypad4,
-             kVK_ANSI_Keypad5, kVK_ANSI_Keypad6, kVK_ANSI_Keypad7, kVK_ANSI_Keypad8,
-             kVK_ANSI_Keypad9:
-            guard flags == .command else { return false }
-            fallthrough
-        case kVK_Tab where [[.control], [.control, .shift]].contains(flags),
-             kVK_ANSI_N where flags == .command,
-             kVK_ANSI_W where flags.contains(.command),
-             kVK_ANSI_T where [[.command], [.command, .shift]].contains(flags),
-             kVK_ANSI_Q where flags == .command,
-             kVK_ANSI_R where flags == .command:
-            guard view.window?.firstResponder is WebView else { return false }
-            NSApp.menu?.performKeyEquivalent(with: event)
-            return true
+        // Handle tab switching (CMD+1 through CMD+9)
+        if flags == .command, "123456789".contains(key) {
+            if isWebViewFocused {
+                NSApp.menu?.performKeyEquivalent(with: event)
+                return true
+            }
+            return false
+        }
 
-        case kVK_ANSI_Y where flags == .command:
-            if NSApp.delegateTyped.featureFlagger.isFeatureOn(.historyView) {
-                return false
+        // Handle browser tab/window actions
+        if isWebViewFocused {
+            switch (key, flags, flags.contains(.command)) {
+            case ("n", [.command], _),
+                 ("t", [.command], _), ("t", [.command, .shift], _),
+                 ("w", _, true),
+                 ("q", [.command], _),
+                 ("r", [.command], _):
+                NSApp.menu?.performKeyEquivalent(with: event)
+                return true
+
+            case ("tab", [.control], _),
```

Comment:
@tomasstrba, this handler isn‘t working, it either needs to be `"\t"` or `kVK_Tab`

Reply:
Come on, GPT! Good catch :thus far

Reply:
Resolved ✔️